applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -227,18 +229,41 @@ int poweron_uart(void)
 	return err;
 }
 
-int set_uart_baudrate(uint32_t baudrate)
+int slm_uart_configure(uint32_t baudrate, uint8_t hwfc)
 {
 	int err;
 
-	LOG_DBG("Set uart baudrate to: %d", baudrate);
+	LOG_DBG("Set uart baudrate to: %d, hw flow control %d", baudrate, hwfc);
 
 	slm_uart.baudrate = baudrate;
+	slm_uart.flow_ctrl = hwfc;
 	err = uart_configure(uart_dev, &slm_uart);
 	if (err != 0) {
 		LOG_ERR("uart_configure: %d", err);
 	}
-
+/* Set HWFC dynamically */
+#if defined(CONFIG_UART_0_NRF_HW_ASYNC_TIMER)
+	if (hwfc) {
+		nrf_uarte_hwfc_pins_set(NRF_UARTE0,
+					DT_PROP(DT_NODELABEL(uart0), rts_pin),
+					DT_PROP(DT_NODELABEL(uart0), cts_pin));
+	} else {
+		nrf_uarte_hwfc_pins_disconnect(NRF_UARTE0);
+		nrf_gpio_pin_clear(DT_PROP(DT_NODELABEL(uart0), rts_pin));
+		nrf_gpio_cfg_output(DT_PROP(DT_NODELABEL(uart0), rts_pin));
+	}
+#endif
+#if defined(CONFIG_UART_2_NRF_HW_ASYNC_TIMER)
+	if (hwfc) {
+		nrf_uarte_hwfc_pins_set(NRF_UARTE2,
+					DT_PROP(DT_NODELABEL(uart2), rts_pin),
+					DT_PROP(DT_NODELABEL(uart2), cts_pin));
+	} else {
+		nrf_uarte_hwfc_pins_disconnect(NRF_UARTE2);
+		nrf_gpio_pin_clear(DT_PROP(DT_NODELABEL(uart2), rts_pin));
+		nrf_gpio_cfg_output(DT_PROP(DT_NODELABEL(uart2), rts_pin));
+	}
+#endif
```

Comment:
Above logic seems to be Honeywell-only spec (after HWFC adjustment).
Could you double-check with Larry?